conda/cli/conda_argparse.py

```diff
@@ -843,41 +869,37 @@ def configure_parser_list(sub_parsers):
     p.add_argument(
         '-c', "--canonical",
         action="store_true",
-        help="Output canonical names of packages only. Implies --no-pip. ",
+        help="Output canonical names of packages only.",
     )
     p.add_argument(
         '-f', "--full-name",
         action="store_true",
-        help="Only search for full names, i.e., ^<regex>$.",
+        help="Only search for full names, i.e., ^<regex>$. "
+             "--full-name NAME is identical to regex '^NAME$'.",
     )
     p.add_argument(
         "--explicit",
         action="store_true",
-        help="List explicitly all installed conda packaged with URL "
+        help="List explicitly all installed conda packages with URL "
              "(output may be used by conda create --file).",
     )
     p.add_argument(
         "--md5",
         action="store_true",
-        help="Add MD5 hashsum when using --explicit",
+        help="Add MD5 hashsum when using --explicit.",
     )
     p.add_argument(
         '-e', "--export",
         action="store_true",
-        help="Output requirement string only (output may be used by "
-             " conda create --file).",
+        help="Output explicit, machine-readable requirement strings instead of "
+             "human-readable lists of packages. This output may be used by "
+             "conda create --file.",
     )
     p.add_argument(
         '-r', "--revisions",
         action="store_true",
-        help="List the revision history and exit.",
+        help="List the revision history.",
     )
-    p.add_argument(
-        "--no-pip",
-        action="store_false",
-        default=True,
-        dest="pip",
-        help="Do not include pip-only installed packages.")
```

Comment:
Should this be removed?

Reply:
Yeah, I wasn't totally sure on that. @kenodegard said it should be deprecated as an argument, so maybe we should just mark it as (deprecated) in here? Or I'll make a ticket to actually properly deprecate the argument and the argparse work can be done then?

Reply:
This argument is unused

<details>
<summary>`--no-pip` → `args.pip` → `piplist` and this `piplist` variable is unused</summary>

https://github.com/conda/conda/blob/8b846957c4338188706959756e7dad1619727c59/conda/cli/conda_argparse.py#L875-L880

https://github.com/conda/conda/blob/8b846957c4338188706959756e7dad1619727c59/conda/cli/main_list.py#L139-L141

https://github.com/conda/conda/blob/8b846957c4338188706959756e7dad1619727c59/conda/cli/main_list.py#L68-L86

</details>

But I agree it should go through a deprecation process, not removed outright

Reply:
Gotcha gotcha. Will fix!